data-table-row.html

```diff
@@ -82,20 +83,33 @@
       _onTap: function(e) {
         var path = Polymer.dom(e).path;
 
-        if (path && path[0].tagName === "INPUT") {
-          e.stopPropagation();
-        } else {
-          //TODO: event should be fired only when template exists
-          var expandEvent = this.fire('iron-data-table-row-expanding', { row: this, target: e.target }, { bubbles: true, cancelable: true });
-          if (this.rowDetailTemplate && !expandEvent.defaultPrevented) {
-            this.expandDetails = !this.expandDetails;
-          }
-
-          //TODO: event should be fired only when selection is enabled
-          var selectEvent = this.fire('iron-data-table-row-selecting', { row: this, target: e.target }, { bubbles: true, cancelable: true });
-          if (selectEvent.defaultPrevented) {
+        if (path) {
```

Comment:
I understand that whoever uses `iron-data-table` can add a listener for `iron-data-table-row-expanding` and/or `iron-data-table-row-selecting` and chose when to disable the default handler here. But providing that functionality using an attribute seems easier for users of `iron-data-table`. Isn't that really the essence of web-components? IMHO, this way, there's a balance between abstraction and extensibility.